enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,270 @@
+
+To distinguish default _no-config_ backups from user configured counterparts, a `config.openshift.io/v1alpha1 Backup` CR's name must be `default`.
+
+Initially, the sidecar container within etcd pods is not running (i.e. disabled). However, creating a `Backup` CR with `name=default` enables the container.
+
+Once enabled, the backups are being taken on each master node according to the provided `schedule` within the CR. Moreover, the backups are being maintained according to the supplied retention policy. 
+
+The default CR's configuration could be overridden by the user according to their preference.
+
+The backups are being stored within a `hostPath` volume. The volume is a directory within each master node's file system.
+
+Before deciding upon this approach, several approaches that have been investigated. See [Alternatives](#Alternatives).
+
+### Open Questions
+
+Currently, the no-config backup sidecar container exits upon error. Since the container is being deployed within the etcd Pod, the whole Pod will never be `Running` as long as the container is failing.
+Once the container fails, it will be automatically restarted according to exponential backoff strategy. This scenario leads to `CrashLoopBackOff` and the etcd Pod will never be `Running`. 
+
+As the no-config backup sidecar is being deployed alongside each etcd member, consequently an error within this container leads to unavailable etcd cluster. Hence, failing the container upon error is not a viable option.
+Also disabling the container from the etcd pod upon error is not recommended, unless requested by the user. As discussed above, enabling and disabling the sidecar container leads to deploying a new version of the static pods.
+This scenario leads to a duration of cluster unavailability. 
+
+As shown above, it is necessary to report the no-config backup sidecar failure without failing the container. Below are possible resolution scenarios. 
+
+- Log the error upon failure
+  - A log parser could be used in order to detect the error, without failing the container.
+  
+- Update Cluster-etcd-Operator StatusConditions
+  - Reporting the condition and degrading the operator could be a solution. 
+
+- Use a Prometheus gauge in order to report the container situation. 
+
+### Implementation Details/Notes/Constraints
+
+The _no-config_ backups builds atop of the existing [automated backup of etcd](https://docs.openshift.com/container-platform/4.15/backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.html#creating-automated-etcd-backups_backup-etcd).
+
+The `PeriodicBackupController` within **cluster-etcd-operator** checks for `Backup` CR with `name=default` within every sync. Upon applying a `Backup` CR with `name=default`, the _no-config_ backup sidecar container is being enabled within each etcd pod in Openshift cluster.
```

Comment:
This just seems like a special case within the periodic backups no? Why do we need a new mechanism rather than the periodic backup controller just handling running backups on a default schedule?

Reply:
good question 👍🏽  

So, the `default` CR is the same type of CR (i.e. config.openshift.io/v1alpha1 Backup).  The reasons to use `PeriodicBackupController` are 

- Having two separate controllers to monitor the same CR (config.openshift.io/v1alpha1 Backup) type is `anti-pattern` 
- If the user Creates or Deletes the default no-config backup, it should not influence the Cron based Backups, as they are orthogonal.
- Having the same controller manages the all the Backup CRs is ideal for `FeatureGate` reasons.

Reply:
I appreciate the idea to use the same controller, that makes sense, and I think my question is agreeing with what you've said.

My point here was that, this part of the EP describes a mechanism for initiating backups. I'm trying to work out how this compares to the mechanism in use for the periodic backups.

As in, how are the periodic backups implemented, do they use a set of static pods?

And then from that, if there is a difference in how the backup is implemented for default vs periodic, why do we need that difference?

Reply:
> As in, how are the periodic backups implemented, do they use a set of static pods?

No, they are not. So the high-level design is being depicted [here](https://docs.google.com/document/d/1iIyJYAl2G70RPsPY3wF65i06PGFWBJs-Hk7z68tFPEM/edit?usp=sharing) within the `Troubleshooting` section

So `PeriodicBackups` are being implemented using two levels 

- A `OneTime Backup` triggers a Job, which triggers a Pod which runs `backup.sh` script.
- `PeriodicBackups` is a `CronJob` which executes the `OneTime Backup` upon each trigger.

This implementation does not rely on StaticPods at all. 

> And then from that, if there is a difference in how the backup is implemented for default vs periodic, why do we need that difference?


So the main motivations behind `default` method are

- Provide Backups from Day1 without user supplied configurations
- Having the Backups replicated, for disaster recovery scenarios 

iiuc, OCP variants such as `SNO` and `BM` can not use PVC with external storage, the only options are `hostPath` and `localVolume`. 

For cloud based OCP, having a PVC bound to external storage eliminates the need for replicating the backups over control-plane nodes, since the PV is on cloud provider's side, and backup are accessible on Disaster scenarios. 

On the other hand, using `hostPath` on the `default` method is a way to provide backup replicas, while having the same method applicable on `SNO` and `BM`

Reply:
Why do we need `default` instance at all? I imagine we can have multiple instances with different schedules and retention policies (i.e. weekly and monthly snapshots)

Reply:
I actually think these ideas worth exploring 


cc @tjungblu